### PR TITLE
Create TinyMCE command for adding media

### PR DESCRIPTION
**Pull request by dmsnell**
The "add media" dialog appears after clicking on the "add media" icon in
the TinyMCE toolbar.

Previously, this occurred as a direct result of the `onclick` handler in
`editor.addButton()`. Because this meant that adding media was dependent
on clicking the button, it prevented being able to do so in other ways,
such as is the goal in #5071

This patch moves the code to cause the media dialog to appear into a new
command, which the button then calls when it's clicked.

**Testing**

There are no visual or functional changes here. Try to edit a post and add media. Click on the "add media" button. Test any other existing ways you know of to add media into a post to make sure this doesn't introduce any regressions.

**Questions**
 - [x] <del>Is it wrong to name the action the same thing as the button? Do they live in different namespaces or will we inadvertently cause collisions by doing this?</del> - I have renamed the command based on a suggestion from @rodrigoi - it is now `wpcomAddMedia`

cc: @aduth

### CODE REVIEW

**Comment by rodrigoi** [comment]
Code looks good and I didn't find any regressions.

There is nothing on TinyMCE's documentation warning about collisions between buttons and commands. I don't think it will be an issue.
What could be an issue is that we have two different styles when naming commands and buttons. I personally like `wpcomAddMedia` for commands and `wpcom_add_media` for buttons. It may be a good idea to choose a style, considering that we'll be adding many commands and buttons to the widget drop down in the near future. Thoughts @aduth ?

**Comment by dmsnell** [comment]
> I personally like wpcomAddMedia for commands and wpcom_add_media for buttons.

Done! thanks for the suggestion

File: client/components/tinymce/plugins/media/plugin.jsx [hunk 1]
```diff
@@ -281,28 +281,28 @@ function mediaButton( editor ) {
 		renderDropZone( { visible: event.type === 'dragend' } );
 	}
 
+	editor.addCommand( 'wpcomAddMedia', () => {
+		var selectedSite = sites.getSelectedSite();
```
Comment (gwwar) [inline client/components/tinymce/plugins/media/plugin.jsx]: Any reason for the var over a const?

**Comment by gwwar** [comment]
Is this supposed to be behind a feature flag?

I see the new edit option
screen shot 2016-04-28 at 4 47 27 pm
and a temp modal
screen shot 2016-04-28 at 4 46 43 pm

File: client/components/tinymce/plugins/media/plugin.jsx (hunk 1 above)
Comment (dmsnell) [inline client/components/tinymce/plugins/media/plugin.jsx]: @gwwar just because I was trying not to change the code

you think I should change it?

Comment (gwwar) [inline client/components/tinymce/plugins/media/plugin.jsx]: There's already mixed usage in the file, so either way works with me. :)

**Comment by dmsnell** [comment]
> Is this supposed to be behind a feature flag?

Strange @gwwar - I see this behavior in *master*

@aduth - any thoughts?

**Comment by aduth** [comment]
>I see the new edit option

@gwwar @dmsnell This is an in-progress feature related to #307, [feature-flagged by `post-editor/media-advanced`](https://github.com/Automattic/wp-calypso/blob/595832734d193fca6b166975c76966c481dc5325/client/components/tinymce/plugins/media/advanced/index.jsx#L21-L23) (development environments only).

File: client/components/tinymce/plugins/media/plugin.jsx (hunk 1 above)
Comment (aduth) [inline client/components/tinymce/plugins/media/plugin.jsx]: Would be fine to change I think.

**Comment by aduth** [comment]
Looks good 👍

**Pull request merged**